fiware-rss/src/test/java/es/upm/fiware/rss/service/SettlementManagerTest.java

```diff
@@ -268,6 +266,83 @@ public void testRunSettlementAll() throws RSSException {
         verify(poolManager).closeTaskPool(callbackUrl);
     }
 
+    private SharingReport mockSharingReport(int id, boolean paid) {
+        BmCurrency curr = Mockito.mock(BmCurrency.class);
+        when(curr.getTxIso4217Code()).thenReturn("EUR");
+
+        DbeAggregator aggr = Mockito.mock(DbeAggregator.class);
+        when(aggr.getTxEmail()).thenReturn(aggregatorId);
+
+        DbeAppProviderId provId = Mockito.mock(DbeAppProviderId.class);
+        when(provId.getAggregator()).thenReturn(aggr);
+        when(provId.getTxAppProviderId()).thenReturn(providerId);
+
+        DbeAppProvider provider = Mockito.mock(DbeAppProvider.class);
+        when(provider.getId()).thenReturn(provId);
+
+        SharingReport report = Mockito.mock(SharingReport.class);
+        when(report.getId()).thenReturn(id);
+
+        when(report.getOwner()).thenReturn(provider);
+
+        when(report.getAggregatorValue()).thenReturn(new BigDecimal(0.3));
+        when(report.getAlgorithmType()).thenReturn("FIXED_PERCENTAGE");
+        when(report.getCurrency()).thenReturn(curr);
+        when(report.getOwnerValue()).thenReturn(new BigDecimal(0.7));
+        when(report.getProductClass()).thenReturn(productClass);
+        when(report.getDate()).thenReturn(new Date());
+        when(report.getPaid()).thenReturn(paid);
+        return report;
+    }
+
+    private List<SharingReport> generateReports(int paid, int notpaid) {
+        if (paid < 0 || notpaid < 0) {
+            return new ArrayList<>();
+        }
+
+        return IntStream.range(0, paid + notpaid).mapToObj(id -> {
+            return mockSharingReport(id, id + 1 <= paid);
+        }).collect(Collectors.toList());
+    }
+
+    @Test
+    public void testGetSharingReportsEmpty() {
+        Optional<List<SharingReport>> sreports1 = Optional.empty();
+        when(sharingReportDao.getSharingReportsByParameters(aggregatorId, providerId, productClass, true, 0, -1))
+                .thenReturn(sreports1);
+
+        List<RSSReport> reports1 = toTest.getSharingReports(aggregatorId, providerId, productClass, true, 0, -1);
+
+        Assert.assertEquals(0, reports1.size());
+    }
+
+    @Test public void testGetReports() {
+        List<SharingReport> sharingReportList = generateReports(1, 1);
+        Optional<List<SharingReport>> sharingReportListOpt = Optional.of(sharingReportList);
+        when(sharingReportDao.getSharingReportsByParameters(aggregatorId, providerId, productClass, true, 0, -1))
+                .thenReturn(sharingReportListOpt);
+
+        List<RSSReport> rssReportsList = toTest.getSharingReports(aggregatorId, providerId, productClass, true, 0, -1);
+        Assert.assertEquals(2, rssReportsList.size());
+
+        IntStream.range(0, 2).forEach(i -> {
```

Comment:
In this case, I think that you can stream the rssReportsList itself and then apply a lambda function that receives as parameter the concrete element of the list, instead of generating a range of indexes.

rssReportsList.stream().forEach(rss -> {...